src/components/GlobalPersistGateHydrationListener.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  selectisPersistGateHydrationInProgress,
+  setIsPersistGateHydrationComplete,
+  setIsPersistGateHydrationInProgress,
+} from '../redux/slices/persistGateHydration';
+
+const PERSIST_GATE_HYDRATION_DURATION_MS = 50; // This number is mostly arbitrary. Long enough to ensure that the hydration is complete.
+
+// This component listens to the REHYDRATE event from redux persist
+// and dispatches an action to set the hydration as complete.
+// The component works by setting a timeout to fire after the REHYDRATE event.
+// Because the redux store is synchronous, there's no way to ensure the order of actions.
+// So we have to use a timeout to ensure that the hydration complete event happens *after* REHYDRATE.
+const GlobalPersistGateHydrationListener = () => {
+  const dispatch = useDispatch();
+  const isPersistGateHydrationInProgress = useSelector(selectisPersistGateHydrationInProgress);
+
+  useEffect(() => {
+    if (isPersistGateHydrationInProgress) {
+      setTimeout(() => {
+        dispatch({ type: setIsPersistGateHydrationComplete.type, payload: true });
+        dispatch({ type: setIsPersistGateHydrationInProgress.type, payload: false });
+      }, PERSIST_GATE_HYDRATION_DURATION_MS);
+    }
```

Comment:
I believe we should add `useEffect`'s cleanup function to clear the internal.

```
return () => {
        clearTimeout(handler);
      };
```

You can refer to `src/hooks/useDebounce.ts` for an example.

Reply:
We don't want to clear the timeout in this case. We always want the dispatch action to go through.